### ISSUE TITLE
Compatibility issue of LiquidCore-0.6.1 with xcode-10.1 and older versions.

### ISSUE THREAD

**Issue by A-P-Singh** [LiquidPlayer/LiquidCore#101]
Hi , 
Thanks for the new release of 0.6.1 with all fixes till now what ever we faced .

Now we are facing new - When we try to run xcode-10.1 with Liquidcore-0.6.1 we are getting some issue .
If we comment below line of code from xcode-10.1 then its working fine . But I believe this is not  the proper solution .
Need to comment code: 
Filename: value.cpp

<img width="700" alt="Screen Shot 2019-05-09 at 1 22 15 PM" src="https://user-images.githubusercontent.com/22491079/57436638-b96d8c00-725d-11e9-9696-dca9ea18198e.png">

Good new is every thing is working fine in xcode-10.2.1 .
But In our project we are using 10.1 . which is giving above issue . 
Please suggest the proper solution
Thanks
Ajay[](url)

**Comment by ericwlange** [comment]
Yes, kJSTypeSymbol was added in 10.2. This needs to be wrapped in a
conditional compilation macro. I tried briefly to figure out which #define
is set to identify the xcode version but didn't immediately find it. If you
figure it out please submit a PR or let me know and I will add it.

**Comment by A-P-Singh** [comment]
Hi @ericwlange ,
I am not much familiar with c++ language. So Can you fix this or If you want me to do that then it will take some more time .

**Comment by ericwlange** [comment]
Can you try the following patch and let me know if it works?

```
diff --git a/LiquidCoreiOS/LiquidCore/V82JSC/Value.cpp b/LiquidCoreiOS/LiquidCore/V82JSC/Value.cpp
index aa50e75e4..ba97bda14 100644
--- a/LiquidCoreiOS/LiquidCore/V82JSC/Value.cpp
+++ b/LiquidCoreiOS/LiquidCore/V82JSC/Value.cpp
@@ -4,6 +4,7 @@
  * Distributed under the MIT License.  See LICENSE.md at
  * https://github.com/LiquidPlayer/LiquidCore for terms and conditions.
  */
+#include <Availability.h>
 #include "V82JSC.h"
 #include "JSObjectRefPrivate.h"
 #include "Value.h"
@@ -102,10 +103,12 @@ Local<v8::Value> Value::New(const Context *ctx, JSValueRef value, BaseMap *map)
                 }
                 break;
             }
+#if __IPHONE_OS_VERSION_MAX_ALLOWED > __IPHONE_12_1
             case kJSTypeSymbol: {
                 map = isolateimpl->m_symbol_map;
                 break;
             }
+#endif
             default:
                 break;
         }
```

**Comment by A-P-Singh** [comment]
If I run below code then its working on xcode-10.0 and 10.1 . But its not working on xcode-10.2 .
I think we are very close to the solution.

<img width="877" alt="Screenshot 2019-05-10 at 4 56 26 PM" src="https://user-images.githubusercontent.com/22491079/57524275-f95b6e80-7344-11e9-88d4-2a639f4853dc.png">


Please suggest something which can fix xcode-10.2 issue also

**Comment by A-P-Singh** [comment]
Second thing ,
If I run      
   #if __IPHONE_OS_VERSION_MAX_ALLOWED > __IPHONE_12_1

Then its not working on 10.0 .

Regards
Ajay

**Comment by ericwlange** [comment]
Ok, I think this may be the winner.  Try:

```
#ifdef __IPHONE_12_2
```

**Comment by A-P-Singh** [comment]
Thanks @ericwlange , its working fine for all three (10,10.1 and 10.2.1) version of xcode .

But I think I don't have access of your liquidcore repo . So when I am trying to  push the code in my feature branch then I am getting some error .
<img width="835" alt="Screenshot 2019-05-10 at 2 57 39 PM" src="https://user-images.githubusercontent.com/22491079/57597711-1a52d800-756e-11e9-8c01-130d2f9e9c6f.png">

Please give me the access .

**Comment by ericwlange** [comment]
You won't be able to push directly to the LiquidCore repo.  You would have to submit a PR.

But if that's the only change required, I can commit it.

**Comment by ericwlange** [comment]
This has been committed.  You can use:

```ruby
pod 'LiquidCore', :git => 'https://github.com/LiquidPlayer/LiquidCore.git'
```

to use the bleeding edge version until a new release has been pushed.

**Comment by A-P-Singh** [comment]
Thanks @ericwlange 
When can we expect new version Liquidcore with fix of this issue ?

**Comment by ericwlange** [comment]
I don't plan to push another release just for this small change for older SDKs.  There is too much personal overhead.  The next release, 0.7.0, will feature an upgrade to Node 10.15.3 and there is still a lot of work ahead of me to accomplish that, so it is not going to be anytime soon.

What I recommend until that happens, is that you use:

```ruby
pod 'LiquidCore', :git => 'https://github.com/LiquidPlayer/LiquidCore.git', :commit => '7dc9c3e'
```

to ensure you only get that change.

**Comment by umangkathuria** [comment]
@ericwlange 
While using this branch with iOS deployment version 9.0, it appears that the pod install fails.

<img width="767" alt="Screenshot 2019-05-30 at 5 34 53 PM" src="https://user-images.githubusercontent.com/13998498/58631818-540f3700-8301-11e9-8919-5fffd74059ae.png">


Above screenshot shows the exact error I am getting.
It worked with iOS version 12.0

Does the library not support lower versions ?

**Comment by ericwlange** [comment]
I have not tried it with targets less than 10.0.  You can try it by changing [line 14 of `LiquidCore.podspec`](https://github.com/LiquidPlayer/LiquidCore/blob/master/LiquidCore.podspec#L14) and see if it works.  The biggest concern is which version of JavaScriptCore exists on older devices.  V82JSC, the layer that dynamically translates V8 API calls to JavaScriptCore depends on ES6 features.  It should work fine so long as the version is not too old.  But I've never tried.  I just know that 10+ works.  If you test it and it works, let me know and I will lower the version.

**Comment by ericwlange** [comment]
This has been fixed in 0.6.2.  LiquidCore will not support iOS versions less than 11.0.
